cns/imds/client.go

```diff
@@ -102,10 +104,50 @@ func (c *Client) GetVMUniqueID(ctx context.Context) (string, error) {
 	return vmUniqueID, nil
 }
 
-func (c *Client) getInstanceComputeMetadata(ctx context.Context) (map[string]any, error) {
+func (c *Client) GetNCVersionsFromIMDS(ctx context.Context) (map[string]string, error) {
```

Comment:
Naming: `*FromIMDS` is implied since this is an IMDS client.

```suggestion
func (c *Client) GetNCVersions(ctx context.Context) (map[string]string, error) {
```